packages/client/hmi-client/src/workflow/ops/model-transformer/tera-model-transformer.vue

```diff
@@ -0,0 +1,114 @@
+	<div class="background">
+		<Suspense>
+			<tera-model-jupyter-panel
+				asset-id="bcd32c48-1daa-4a52-8ed7-147c1dc4b0b0"
+				:model-configuration-id="modelConfigurationId"
+				:project="props.project"
+				:model="null"
+				:show-kernels="false"
+				:show-chat-thoughts="false"
+				@new-model-saved="addOutputPort"
+				:notebook-session="notebookSession"
+			/>
+		</Suspense>
+	</div>
+</template>
+
+<script setup lang="ts">
+// Proxy to use tera-dataset via a workflow context
+
+import { IProject } from '@/types/Project';
+import { WorkflowNode, WorkflowPortStatus } from '@/types/workflow';
+import TeraModelJupyterPanel from '@/components/model/tera-model-jupyter-panel.vue';
+import { computed, onMounted, ref } from 'vue';
+import { workflowEventBus } from '@/services/workflow';
+import { createNotebookSession, getNotebookSessionById } from '@/services/notebook-session';
+import { v4 as uuidv4 } from 'uuid';
+import { NotebookSession } from '@/types/Types';
+import { cloneDeep } from 'lodash';
+import { getModel, getModelConfigurations } from '@/services/model';
+import { addDefaultConfiguration } from '@/services/model-configurations';
+
+const props = defineProps<{
+	node: WorkflowNode;
+	project: IProject;
+}>();
+const modelConfigurationId = computed(() => {
+	// for now we are only using 1 model configuration for the llm at a time, this can be expanded in the future
+	const modelConfirgurationList = props.node?.inputs
+		.filter((inputNode) => inputNode.status === WorkflowPortStatus.CONNECTED && inputNode.value)
+		.map((inputNode) => inputNode.value![0]);
+
+	return modelConfirgurationList[0];
+});
+
+const notebookSession = ref(<NotebookSession | undefined>undefined);
+
+onMounted(async () => {
+	let notebookSessionId = props.node.state?.notebookSessionId;
+	if (!notebookSessionId) {
+		// create a new notebook session log if it does not exist
+		const response = await createNotebookSession({
+			id: uuidv4(),
+			name: props.node.id,
+			description: '',
+			data: { history: [] },
+			timestamp: new Date().toISOString()
+		});
+		notebookSessionId = response?.id;
+
+		if (notebookSessionId) {
+			// update the node state with the notebook session id
+			const state = cloneDeep(props.node.state);
+			state.notebookSessionId = notebookSessionId;
+			workflowEventBus.emit('update-state', {
+				node: props.node,
+				state
+			});
+		}
+	}
+
+	notebookSession.value = await getNotebookSessionById(notebookSessionId);
+});
+
+const addOutputPort = async (data) => {
+	if (!data.id) return;
+	// get model
+	const model = await getModel(data.id);
+
+	if (!model) return;
+
+	const state = cloneDeep(props.node.state);
+	state.modelId = model?.id;
+	// update node state with the model id
+	workflowEventBus.emit('update-state', { node: props.node, state });
+
+	// set default configuration
+	await addDefaultConfiguration(model);
+
+	// setting timeout...elastic search might not update default config in time
+	await setTimeout(async () => {
```

Comment:
This should be fixed for document creation in ES, can we verify that this is working without the timeout?

Reply:
hmm - ill test again, I was having issues where I wasn't getting the default config back. Will test without

Reply:
did a few tests, unfortunately it looks like the timeout is needed!

Reply:
okay ... we will come back to this later